src/components/Chat/ChatItem.tsx

```diff
@@ -0,0 +1,130 @@
+import * as React from 'react'
+import * as PropTypes from 'prop-types'
+import * as cx from 'classnames'
+
+import {
+  createShorthandFactory,
+  customPropTypes,
+  IRenderResultConfig,
+  UIComponent,
+} from '../../lib'
+import {
+  ComponentPartStyle,
+  ComponentVariablesInput,
+  IComponentPartStylesPrepared,
+} from '../../../types/theme'
+import { Extendable, ItemShorthand, ReactChildren } from '../../../types/utils'
+import ChatBubble from './ChatBubble'
+import ChatAction from './ChatAction'
+import Divider from '../Divider/Divider'
+import childrenExist from '../../lib/childrenExist'
+
+export interface IChatItemProps {
+  action?: ItemShorthand
+  as?: any
+  bubble?: ItemShorthand
+  children?: ReactChildren
+  className?: string
+  divider?: ItemShorthand
+  styles?: ComponentPartStyle
+  variables?: ComponentVariablesInput
+}
+
+class ChatItem extends UIComponent<Extendable<IChatItemProps>, any> {
+  static className = 'ui-chat__item'
+
+  static create: Function
+
+  static displayName = 'ChatItem'
+
+  static propTypes = {
+    as: customPropTypes.as,
+
+    /** Shorthand for the bubble message. */
```

Comment:
Since `bubble`, `divider` and `action` props are mutually exclusive, we need a better way to prevent users try more than one at a time; I'd suggest something like this:
```typescript
/** Shorthand for the bubble message. */
bubble: customPropTypes.disallow(['action', 'divider']),

/** Shorthand for the divider message. */
divider: customPropTypes.disallow(['action', 'bubble']),

/** Shorthand for the control message. */
action: customPropTypes.disallow(['bubble', 'divider']),
```

Reply:
Just the thing I was missing. Thanks a lot for the suggestion Alex!